presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestPrestoContainerBasicQueries.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.nativeworker;
+
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.Container;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+public class TestPrestoContainerBasicQueries
+{
+    private static final String PRESTO_COORDINATOR_IMAGE = System.getProperty("coordinatorImage", "presto-coordinator:latest");
+    private static final String PRESTO_WORKER_IMAGE = System.getProperty("workerImage", "presto-worker:latest");
+    private static final String BASE_DIR = System.getProperty("user.dir");
+    private static final Network network = Network.newNetwork();
+    private GenericContainer<?> coordinator;
+    private GenericContainer<?> worker;
+
+    @BeforeClass
+    public void setUp()
+            throws InterruptedException
+    {
+        coordinator = new GenericContainer<>(PRESTO_COORDINATOR_IMAGE)
+                .withExposedPorts(8081)
+                .withNetwork(network).withNetworkAliases("presto-coordinator")
+                .withFileSystemBind(BASE_DIR + "/testcontainers/coordinator/etc", "/opt/presto-server/etc", BindMode.READ_WRITE)
+                .withFileSystemBind(BASE_DIR + "/testcontainers/coordinator/entrypoint.sh", "/opt/entrypoint.sh", BindMode.READ_ONLY)
+                .waitingFor(Wait.forLogMessage(".*======== SERVER STARTED ========.*", 1))
+                .withStartupTimeout(Duration.ofSeconds(120));
+
+        worker = new GenericContainer<>(PRESTO_WORKER_IMAGE)
```

Comment:
Many times developers have set a etc configuration in presto-native-execution/etc. This set of config files are used when they start presto-server for worker on their local machines. Can these files be reused if possible ?

Reply:
The minimum configurations required for worker to run the functional tests  are kept in `presto-native-execution/testcontainers/nativeworker/velox-etc`. The developers adding the new functional tests shall be responsible for push the required configurations also to this folder, otherwise function tests will fail in pipeline once enabled.

Reply:
Hmm... Seems like we should abstract some utilities for this code so that engineers can write multiple tests each with different config.